### PR TITLE
Add backend support matrices for different platforms

### PR DESCRIPTION
**Pull request by tanmayv25**
@wphicks  Can you verify the accuracy for FIL backend support across platforms?
@szalpal  Can you do the same for DALI backend?

### CODE REVIEW

File: docs/backend_platform_support_matrix.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+Note that not all Triton backends support all these different
+platforms. The purpose of this document is to go over what all
+compute platforms are supported by each of these Triton backends.
+GPU in this document refers to Nvidia GPU. See
+[GPU, Driver, and CUDA Support Matrix](https://docs.nvidia.com/deeplearning/frameworks/support-matrix/index.html)
+to learn more about supported GPUs.
+
+## Ubuntu 20.04
+
+The table below describes target device supported for inference by
```
Comment (Tabrizian) [inline docs/backend_platform_support_matrix.md:43]: device -> device(s)

File: docs/backend_platform_support_matrix.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+| TensorRT     |    GPU    |     GPU       |  Unsupported |  Unsupported      |
+| ONNX Runtime |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          |
+| TensorFlow   |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          | 
+| PyTorch      |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          |
+| OpenVINO     |    CPU    |     CPU       |     CPU      |      CPU          |
+| Python       |CPU/GPU[^1]|  CPU/GPU[^1]  |     CPU[^1]  |      CPU[^1]      |
+| DALI         |   GPU     |     GPU       |  Unsupported |  Unsupported      |
+| FIL          |  CPU/GPU  |  Unsupported  |    CPU       |  Unsupported      |
+
+
```
Comment (Tabrizian) [inline docs/backend_platform_support_matrix.md:57]: I think it would be good to add the definition of CPU-only too (i.e. a container that doesn't have the CUDA libraries).

File: docs/backend_platform_support_matrix.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+## Windows 10
+
+Only TensorRT and ONNX Runtime backends are supported on Windows.
+
+| Backend      | x86-GPUs  | x86 CPU only |
```
Comment (Tabrizian) [inline docs/backend_platform_support_matrix.md:63]: Do we release the x86 CPU only version or this is just referring to building the windows version of Triton using build.py with CPU-only flag?

File: docs/backend_platform_support_matrix.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+
+## AWS Inferentia
+
+Currently, inference on AWS Inferentia is only supported via
+[python backend](https://github.com/triton-inference-server/python_backend#running-with-inferentia)
+where the deployed python script invokes AWS Neuron SDK.
+
+
+[^1]: The supported devices for python backend are mentioned with
```
Comment (Tabrizian) [inline docs/backend_platform_support_matrix.md:91]: * python backend -> Python Backend
* python -> Python

**Comment by wphicks** [comment]
FIL info is correct.

**Review by wphicks — DISMISSED** [review]

File: docs/backend_platform_support_matrix.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Backend-Platform Support Matrix
+
+Even though Triton supports inference across varied platforms such as
```
Comment (rmccorm4) [inline docs/backend_platform_support_matrix.md:31]:
```suggestion
Even though Triton supports inference across various platforms such as
```

File: docs/backend_platform_support_matrix.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+| Backend      | x86-GPUs  | ARM-SBSA-GPUs | x86 CPU only | ARM-SBSA CPU only |
+| ------------ | --------- | ------------- | ------------ |------------------ |
+| TensorRT     |    GPU    |     GPU       |  Unsupported |  Unsupported      |
+| ONNX Runtime |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          |
+| TensorFlow   |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          | 
+| PyTorch      |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          |
+| OpenVINO     |    CPU    |     CPU       |     CPU      |      CPU          |
+| Python       |CPU/GPU[^1]|  CPU/GPU[^1]  |     CPU[^1]  |      CPU[^1]      |
+| DALI         |   GPU     |     GPU       |  Unsupported |  Unsupported      |
+| FIL          |  CPU/GPU  |  Unsupported  |    CPU       |  Unsupported      |
```
Comment (rmccorm4) [inline docs/backend_platform_support_matrix.md:55]: Can this table be simplified to just two columns? 

The main question is, **is there an important distinction in the table between "CPU support" and "CPU-only support"** ?

The CPU-only columns with CPU support read a little strange to me.

ex:

| Backend      | x86       | ARM-SBSA      |
| ------------ | --------- | ------------- |
| TensorRT     |  :heavy_check_mark: GPU <br/> :x: CPU | :heavy_check_mark: GPU <br/> :x: CPU       |
| ONNX Runtime |  :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU  |   :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU      |
| TensorFlow   |  :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU  |   :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU      |
| PyTorch      |  :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU  |   :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU      |
| OpenVINO     |  :x: GPU <br/> :heavy_check_mark: CPU    |     :x: GPU <br/> :heavy_check_mark: CPU       |
| Python       |  :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU [^1] |  :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU [^1]  |
| DALI         |  :heavy_check_mark: GPU <br/> :x: CPU     |     :heavy_check_mark: GPU <br/> :x: CPU       |
| FIL          |  :heavy_check_mark: GPU <br/> :heavy_check_mark: CPU  |  Unsupported  |

[^1]: ...

File: docs/backend_platform_support_matrix.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Backend-Platform Support Matrix
+
+Even though Triton supports inference across varied platforms such as
+cloud, data center, edge and embedded devices on NVIDIA GPUs, x86 and
+ARM CPU, or AWS Inferentia, it does so by relying on the backends.
+Note that not all Triton backends support all these different
+platforms. The purpose of this document is to go over what all
```
Comment (rmccorm4) [inline docs/backend_platform_support_matrix.md:35]: "not all ... support all these ..." reads a little weird, maybe this?
```suggestion
Note that not all Triton backends support every
platform. The purpose of this document is to go over what all
```

File: docs/backend_platform_support_matrix.md (hunk 6 above)
Comment (tanmayv25) [reply docs/backend_platform_support_matrix.md:55]: Good idea! I can make this change!

File: docs/backend_platform_support_matrix.md (hunk 2 above)
Comment (tanmayv25) [reply docs/backend_platform_support_matrix.md:57]: With Ryan's suggestion below, this should not be needed I guess. It will simplify our support.

File: docs/backend_platform_support_matrix.md (hunk 3 above)
Comment (tanmayv25) [reply docs/backend_platform_support_matrix.md:63]: I think we can abstract this information with Ryan's suggestion. I am now thinking pointing our CPU only builds should not be a good idea.

**Review by Tabrizian — APPROVED** [review]

**Review by nv-kmcgill53 — APPROVED** [review]
Jetson info is correct.

**Review by rmccorm4 — APPROVED** [review]
Nice doc :rocket:
